### PR TITLE
[vcpkg] Fix termination of vcpkg root search

### PR DESCRIPTION
**Pull request by dg0yt**
**Describe the pull request**
- #### What does your PR fix?  
  Fixes live-lock when not finding `.vcpkg-root`: #22690, #20125.
  (Probably broken since  #11757 when the `FATAL_ERROR` was moved out of the loop.)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  ---

CC @strega-nil-ms

### CODE REVIEW

**Review by faysal247 — COMMENTED** [review]
File commit

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -357,18 +357,23 @@ endif()
...
+while(NOT DEFINED Z_VCPKG_ROOT_DIR)
+    if(EXISTS "${Z_VCPKG_ROOT_DIR_CANDIDATE}/.vcpkg-root")
+        set(Z_VCPKG_ROOT_DIR "${Z_VCPKG_ROOT_DIR_CANDIDATE}" CACHE INTERNAL "Vcpkg root directory")
+    elseif(IS_DIRECTORY "${Z_VCPKG_ROOT_DIR_CANDIDATE}")
         get_filename_component(Z_VCPKG_ROOT_DIR_TEMP "${Z_VCPKG_ROOT_DIR_CANDIDATE}" DIRECTORY)
-        if(Z_VCPKG_ROOT_DIR_TEMP STREQUAL Z_VCPKG_ROOT_DIR_CANDIDATE) # If unchanged, we have reached the root of the drive
-        else()
-            SET(Z_VCPKG_ROOT_DIR_CANDIDATE "${Z_VCPKG_ROOT_DIR_TEMP}")
+        if(Z_VCPKG_ROOT_DIR_TEMP STREQUAL Z_VCPKG_ROOT_DIR_CANDIDATE)
+            break() # If unchanged, we have reached the root of the drive wihout find vcpkg.
```
Comment (vadi2) [inline scripts/buildsystems/vcpkg.cmake:368]:
```suggestion
            break() # If unchanged, we have reached the root of the drive without finding vcpkg.
```

**Comment by JackBoosY** [comment]
I need to confirm this changes with my team members.

**Review by ras0219-msft — COMMENTED** [review]
This LGTM with @vadi2's typo fix. Thanks!

**Pull request merged**